src/utils/ConfigLoader.ts

```diff
@@ -14,11 +15,14 @@ const checkFileExists = (config: dotenv.DotenvConfigOutput) => {
   else return config;
 };
 
+export let ENV: IConfig;
+
 export const loadEnvironmentVariablesFromFiles = () => {
   dotenv.config({ path: DEFAULT_ENV_FILE });
   if (process.env.NODE_ENV) {
     const envFilePath = `${__dirname}/../../.env.${process.env.NODE_ENV}`;
     checkFileExists(dotenv.config({ path: envFilePath }));
+    ENV = new Config();
```

Comment:
It looks like surprising side effect. I guess implementation might not be complete but at this point doing validation in constructor does not look like the right thing.